### PR TITLE
Only post messages to STDIN if subprocess is running

### PR DESCRIPTION
**Pull request by mrmeku**

### CODE REVIEW

**Comment by achew22** [comment]
Can you add an e2e test to validate the absence of the error output?

**Comment by mrmeku** [comment]
> Can you add an e2e test to validate the absence of the error output?

Done! Good thing you asked too, I found a bug in `IsSubprocessRunning` as a result

File: e2e/example_client/BUILD.bazel [hunk 1]
```diff
@@ -13,3 +14,12 @@ go_library(
         "@io_bazel_rules_go//go/tools/bazel_testing:go_default_library",
     ],
 )
+
+go_bazel_test(
```
Comment (achew22) [inline e2e/example_client/BUILD.bazel:18]: The hope was that people would reuse `example_client` as a library and the other tests would live in `e2e/your_test_name` Could I ask you to `git mv` these up a dir? shouldn't be a painful move I hope

File: e2e/example_client/example_client_test.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+	"time"
+)
+
+func TestRestartProcess(t *testing.T) {
+	client := StartLiveReload(t)
+	defer client.Cleanup()
+	client.Kill(t)
+
+	// Wait for the server to kill itself (it has a 1 second delay)
+	time.Sleep(time.Second * 2)
```
Comment (achew22) [inline e2e/example_client/example_client_test.go:15]: Would it be better to in `client.Kill()` try to get the `/config` url over and over again until a connection error? Then there wouldn't need to be a sleep in here

Comment (mrmeku) [reply e2e/example_client/example_client_test.go]: Done

**Comment by mrmeku** [comment]
@achew22 The command unit tests were relying upon the bug in `IsSubprocessRunning` to pass.

I'm not sure how to resolve this without explicitly mocking `process_group.ProcessGroup`

**Pull request closed without merging**